pkg/action/failback.go

```diff
@@ -229,6 +212,161 @@ func (ac *ActionController) deactivateDestinationDuringFailback(action *storkv1.
+			logEvents(msg, "err")
+			action.Status.Status = storkv1.ActionStatusFailed
+			action.Status.Reason = msg
+			ac.updateAction(action)
+			return
+		}
+	}
+
+	_, latestMigration := getLatestMigrationPolicyAndStatus(*referencedMigrationScheduleForLatestMigration)
+
+	// if no migration found, abort the failback process
+	if latestMigration == nil {
+		msg := fmt.Sprintf("No migration found for migrationschedule %s, hence aborting %s operation", referencedMigrationScheduleForLatestMigration.Name, action.Spec.ActionType)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.updateAction(action)
+		return
+	}
+
+	if !isMigrationSuccessful(latestMigration.Status) {
+		msg := fmt.Sprintf("Latest migration for migrationschedule %s/%s has not completed successfully, it's status is %s, hence aborting %s operation", referencedMigrationScheduleForLatestMigration.Namespace, referencedMigrationScheduleForLatestMigration.Name, latestMigration.Status, action.Spec.ActionType)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.updateAction(action)
+		return
+	}
+
+	migrationObject, err := storkClient.GetMigration(latestMigration.Name, migrationSchedule.Namespace)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to get the latest migration %s in migration schedule %s/%s: %v", latestMigration.Name, migrationSchedule.Namespace, migrationSchedule.Name, err)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.updateAction(action)
+		return
+	}
+
+	migrationNamespaces, err := utils.GetMergedNamespacesWithLabelSelector(migrationSchedule.Spec.Template.Spec.Namespaces, migrationSchedule.Spec.Template.Spec.NamespaceSelectors)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to fetch list of namespaces from the MigrationSchedule %s/%s", migrationSchedule.Namespace, migrationSchedule.Name)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.updateAction(action)
+		return
+	}
+	_, actualNamespaces, _ := utils.IsSubList(namespaces, migrationNamespaces)
+
+	err = ac.waitForVolumesBecomeFree(action, migrationObject, actualNamespaces, config)
+	if err != nil {
+		msg := fmt.Sprintf("%v", err)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.updateAction(action)
+		return
+	}
+
+	msg := fmt.Sprintf("Pods using PVCs got scaled down successfully in cluster : %s successful. Moving to the next stage", ac.config.Host)
```

Comment:
ac.config.Host prints the IP address of the master node of kubeconfig's corresponding cluster.

Reply:
However It's printing the kubernetes service's IP address when we try to get ac.config.Host for inClusterConfig i.e. ac.config. For a remote cluster it prints the remote cluster's master node's IP address as expected

Reply:
modified this to config.host . But As @olavangad-px is pointing in case the config is incluster config it may not give a good output, now I am thinking should I remove it ?